README.md

```diff
@@ -1,77 +1,134 @@
+Odo (OpenShift do...) is a CLI tool that provides developers with **fast** and **automated** source code deployments. Odo supplements continuous development by using the power of OpenShift's [Source-to-Image](https://github.com/openshift/source-to-image) with the stableness of [Kubernetes](https://github.com/kubernetes/kubernetes). Developers can immediately start coding while Odo builds, pushes and deploys the application in the background.
 
-## Pre-requisites
-- OpenShift version 3.7.0 and up
+#### Features
 
-To use odo you need access to an OpenShift instance and have OpenShift CLI installed on your local machine (`oc` should be in your $PATH).
+  - **Multiple languages:** Odo supports Node.JS, Ruby, .Net Core, Perl, PHP and Python.
+  - **Speed:** Building your source code *immediately* after saving and deployed to your cluster.
+  - **Reproducible:** Allows for easy reproducibility by using tightly versioned Docker containers for your source code environment.
+  - **Deployability:** Easily deploy a new version, or have Odo automatically build and re-deploy your code on each change.
+  - **Support for multiple components and microservices:** Deploy only what you need. For example, having both a Ruby and a JavaScript application side-by-side.
+  - **Serverless:** No requirement for running a server to automate tasks. Odo talks to OpenShift directly through an API.
+  - **Instant feedback:** Deploy while making edits to files, showing direct and instant feedback.
 
-### OpenShift instance
-You can use [Minishift](https://docs.openshift.org/latest/minishift/index.html) to get a local instance of OpenShift. However odo can be used with any instance of OpenShift.
+#### How Odo works
 
-### OpenShift CLI
-There are different ways to install OpenShift CLI. 
-Please follow [OpenShift documentation](https://docs.openshift.org/latest/cli_reference/get_started_cli.html#installing-the-cli).
+#### Building:
 
-## Installation
-To install `odo` on your system, you can use the fully automated [install.sh](./scripts/install.sh) script.
-This script will enable odo repository on your system and install odo using package manager depending on your system.
-Supported systems are Debian, Ubuntu, Fedora, CentOS and macOS. You can find more information about package repositories in 
-[Advanced installation guide](./docs/advanced-installation-guide.md)
+For dynamic languages such as JavaScript or Ruby, Odo starts with a [builder image](https://github.com/sclorg/s2i-ruby-container), which performs the following steps:
 
-```
-curl -L https://github.com/redhat-developer/odo/raw/master/scripts/install.sh | bash
-```
+  1. Starts the container with the injected source code.
+  2. Transforms the source code into the appropriate runnable setup. For example, installing dependencies such as from a `package.json` or `config.ru` file.
+  3. Builds and deploys the new container behind an http proxy such as Apache.
+
+#### Development:
```

Comment:
I'm still confused about "Building" and "Development" headers/sections. Why is it called like that?